src/actions/mod.rs

```diff
@@ -219,6 +223,51 @@ impl InitActionContext {
         }
     }
 
+    pub fn invalidate_project_model(&self) {
+        *self.project_model.lock().unwrap() = None;
+    }
+
+    pub fn project_model(&self) -> Result<Arc<ProjectModel>, failure::Error> {
+        let cached: Option<Arc<ProjectModel>> = self.project_model.lock().unwrap().clone();
+        match cached {
+            Some(pm) => Ok(pm),
```

Comment:
Here shouldn't we check if the `Cargo.toml` is modified since we get the cache_

Reply:
We forcefuly invalidate project on Cargo.toml changes: https://github.com/rust-lang-nursery/rls/pull/981/files#diff-95d0b21f25a1cd6745c616be6dce85b4R273.

That is, invalidation happens on every Cargo.toml change, but we recompute the project info only when we need it, instead of doing it eagarly.